phobos/sys/traits.d

```diff
@@ -2577,6 +2580,216 @@ template isSameType(T)
     static assert(indexOf!(isSameType!int, Types) == 2);
 }
 
+/++
+    Converts a function type, function pointer type, or delegate type to the
+    corresponding function type.
+
+    For a function, the result is the same as the given type.
+
+    For a function pointer or delegate, the result is the same as it would be
+    for a function with the same return type, the same set of parameters, and
+    the same set of attributes.
+
+    Another way to look at it would be that it's the type that comes from
+    dereferencing the function pointer. And while it's not technically possible
+    to dereference a delegate, it's conceptually the same thing, since a
+    delegate is essentially a fat pointer to a function in the sense that it
+    contains a pointer to a function and a pointer to the function's context.
+    The result of ToFunctionType is the type of that function.
+
+    Note that code which has a symbol which is a function should use
+    $(LREF SymbolType) rather than $(K_TYPEOF) to get the type of the function
+    in order to avoid issues with regards to $(K_PROPERTY) (see the
+    documentation for $(LREF SymbolType) for details).
+
+    See_Also:
+        $(LREF SymbolType)
+  +/
+template ToFunctionType(T)
```

Comment:
Okay so this is normalising the different function types into the symbol variation.

I would suggest a name like ``NormaliseFunctionTypes``.

Using the prefix ``To`` here, suggests that it is coming from something that isn't a function to a function type.
When in fact its coming from a function type to a different function type that is predictable.

Reply:
Well, I guess we'll see what others say, but I don't see how it's any different from `std.conv.to` in the sense that you can convert a type to itself.

`ToFunctionType` says exactly what the trait is doing, and it's much shorter (and avoids issues with American vs British spelling). If anything,

The Phobos v2 version is `FunctionTypeOf`, but it works on both symbols and types, which makes it more complicated (including having to deal with `@property`), and IMHO, that name is not appropriate for a trait which is converting a type rather than getting the type of a symbol (and it's already confusing enough that it can be used for both in std.traits). `ToFunctionType` was the most obvious name that I could think of, and personally, I don't see how `NormalizeFunctionType` is any better.